library/init/meta/expr.lean

```diff
@@ -32,6 +41,9 @@ meta constant expr.macro_def_name (d : macro_def) : name
 meta def expr.mk_var (n : nat) : expr :=
 expr.var (unsigned.of_nat n)
 
+meta constant expr.mk_quote_macro : expr → expr
+meta constant expr.mk_prenum_macro : nat → expr
+meta constant expr.mk_string_macro : string → expr
```

Comment:
Yes, we should. 

> It's a bit weird since we also have expr.mk_macro.

Macros have two different purposes: 

1- Semantic attachments for speeding up type checking (actually this is the original motivation for them). For example: the projection macro will reduce `proj_i (mk a_1 ... a_i ... a_n)` into `a_i` without expanding the recursor (which is quite big for big structures).
I believe we will have more in the future. Example: we may have one for doing resolution in a single step.

2- Annotating pre-terms with extra information.